custom/lib/Auth.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Passage\Client;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Psr7\HttpFactory;
+use Firebase\JWT\JWT;
+use Firebase\JWT\CachedKeySet;
+use OpenAPI\Client\ApiException;
+use Phpfastcache\CacheManager;
+use InvalidArgumentException;
+use UnexpectedValueException;
+use OpenAPI\Client\Configuration;
+use OpenAPI\Client\Api\MagicLinksApi;
+use OpenAPI\Client\Model\CreateMagicLinkRequest;
+use OpenAPI\Client\Model\MagicLink;
+
+class Auth
+{
+    private CachedKeySet $jwks;
+
+    /**
+     * Auth class that provides methods for validating JWTs and creating Magic Links.
+     */
+    public function __construct(private string $appId, private Configuration $config)
+    {
+        $this->appId = $appId;
+        $this->config = $config;
+
+        $httpClient = new Client();
+        $httpFactory = new HttpFactory();
+
+        $cacheItemPool = CacheManager::getInstance('files');
+        $this->jwks = new CachedKeySet(
+            "https://auth.passage.id/v1/apps/{$appId}/.well-known/jwks.json",
+            $httpClient,
+            $httpFactory,
+            $cacheItemPool,
+            null,
+            true
+        );
+    }
+
+    /**
+     * Validate that a JWT is valid and return the Passage user ID associated with the token.
+     *
+     * @param string $jwt The authentication token to be validated
+     *
+     * @return string User ID of the Passage user
+     * @throws InvalidArgumentException JWT format is invalid
```

Comment:
This can also throw an `UnexpectedValueException`.

Reply:
oh thanks 616487a18fb9ce1ef53a84334991bc0bb5387e61